x/issuer/keeper/issuer.go

```diff
@@ -30,10 +30,12 @@ func (k Keeper) UnmarshalIssuer(value []byte) (interface{}, bool) {
 		return types.Issuer{}, false
 	}
 
+	// FIXME: improve these checks
 	//if issuer.Fee == nil {
 	//	return types.Issuer{}, false
 	//}
 
+	// TODO: check issuer DID id is structured correctly
 	if issuer.IssuerDid == "" {
```

Comment:
there is `did.IsValidDID` function that can be used here

Reply:
Great spot => updated in this cascading PR => https://github.com/allinbits/cosmos-cash/pull/253

:smile: